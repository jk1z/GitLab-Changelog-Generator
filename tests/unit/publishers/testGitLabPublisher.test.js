const GitLabPublisher = require("../../../app/publishers/gitlab");
const GitLabAdapter = require("../../../app/adapters/gitlab");
const Logger = require("../../../app/services/logger");
const TagFixture = require("../../fixtures/tag");

jest.mock("../../../app/services/logger");
jest.mock("../../../app/adapters/gitlab");
describe("GitLab publisher", () => {
    describe("#publish", () => {
        let gitLabPublisher;
        let gitlabAdapter;
        let logger;
        beforeEach(() => {
            jest.resetAllMocks();
            gitlabAdapter = new GitLabAdapter();
            logger = new Logger();
            gitLabPublisher = new GitLabPublisher({ gitlabAdapter, loggerService: logger });
        });
        test("should update release note if release note exists", async () => {
            await gitLabPublisher.publish({
                projectId: "123",
                tag: TagFixture.tagDefault(),
                content: `### Release note (2012-05-27)
#### Closed issues
- Consequatur vero maxime deserunt laboriosam est voluptas dolorem. [#6](http://example.com/example/example/issues/6)
#### Merged merge requests
- test1 [#1](http://gitlab.example.com/my-group/my-project/merge_requests/1) ([admin](https://gitlab.example.com/admin))
`
            });
            const loggerInstance = Logger.mock.instances[0];
            expect(loggerInstance.debug).toHaveBeenCalledTimes(1);
            expect(loggerInstance.debug.mock.calls[0][0]).toMatchSnapshot();
            const gitLabAdapterInstance = GitLabAdapter.mock.instances[0];
            expect(gitLabAdapterInstance.updateTagReleaseByProjectIdTagNameAndTagId).toHaveBeenCalledTimes(1);
            expect(gitLabAdapterInstance.updateTagReleaseByProjectIdTagNameAndTagId.mock.calls[0]).toMatchSnapshot();
        });
        test("should create release note if release note is missing", async () => {
            const mockTag = JSON.parse(JSON.stringify(TagFixture.tagDefault()));
            delete mockTag.release;
            await gitLabPublisher.publish({
                projectId: "123",
                tag: mockTag,
                content: `### Release note (2012-05-27)
#### Closed issues
- Consequatur vero maxime deserunt laboriosam est voluptas dolorem. [#6](http://example.com/example/example/issues/6)
#### Merged merge requests
- test1 [#1](http://gitlab.example.com/my-group/my-project/merge_requests/1) ([admin](https://gitlab.example.com/admin))
`
            });
            const loggerInstance = Logger.mock.instances[0];
            expect(loggerInstance.debug).toHaveBeenCalledTimes(1);
            expect(loggerInstance.debug.mock.calls[0][0]).toMatchSnapshot();
            const gitLabAdapterInstance = GitLabAdapter.mock.instances[0];
            expect(gitLabAdapterInstance.createTagReleaseByProjectIdTagNameAndTagId).toHaveBeenCalledTimes(1);
            expect(gitLabAdapterInstance.createTagReleaseByProjectIdTagNameAndTagId.mock.calls[0]).toMatchSnapshot();
        });
    });
});
