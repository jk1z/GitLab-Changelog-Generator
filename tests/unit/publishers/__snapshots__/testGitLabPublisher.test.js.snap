// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitLab publisher #publish should create release note if release note is missing 1`] = `"Creating a new release note"`;

exports[`GitLab publisher #publish should create release note if release note is missing 2`] = `
Array [
  "123",
  "v1.1.0",
  Object {
    "description": "### Release note (2012-05-27)
#### Closed issues
- Consequatur vero maxime deserunt laboriosam est voluptas dolorem. [#6](http://example.com/example/example/issues/6)
#### Merged merge requests
- test1 [#1](http://gitlab.example.com/my-group/my-project/merge_requests/1) ([admin](https://gitlab.example.com/admin))
",
  },
]
`;

exports[`GitLab publisher #publish should update release note if release note exists 1`] = `"Updating the release note"`;

exports[`GitLab publisher #publish should update release note if release note exists 2`] = `
Array [
  "123",
  "v1.1.0",
  Object {
    "description": "### Release note (2012-05-27)
#### Closed issues
- Consequatur vero maxime deserunt laboriosam est voluptas dolorem. [#6](http://example.com/example/example/issues/6)
#### Merged merge requests
- test1 [#1](http://gitlab.example.com/my-group/my-project/merge_requests/1) ([admin](https://gitlab.example.com/admin))
",
  },
]
`;
